### ISSUE TITLE
Column generated_always_as() cannot be used in storage schema

### ISSUE THREAD

**Issue by juandent** [fnc12/sqlite_orm#1020]
Well, it can, but the following happens:
1- if you add a new column in storage that is generated_always_as(), it will be added to the database the first time you call sync_schema()
2- the second time you call sync_schema() it will attempt to insert it AGAIN because pragma.table_info does not return any generated columns present in the database so every time it thinks it is a new column and tries to insert it causing an exception because that column already exists in the database.
3- if you now remove generated_always_as() from the column to make it a regular column the problem ends and the change is  propagated to the database, esentially removing the generated column and replacing it with an ordinary column
4- if you modify the contents of generated_always_as() the column will try to be reinserted and the change will not reach the database
5- no matter what you do, as long as the generated clause is present in the column, it will be tried to be inserted every time sync_schema runs 
6- the only way to stop this is to use a sqlclient such as sqliteStudio and delete the column or make it ordinary (i.e. remove generated_always_as()) and to remove the generated_always_as() from the storage schema for that column

In summary: we cannot define generated_always_as() columns because they will insist in being inserted in the table on every execution of sync_schema().

This happens with current code at the dev branch and also in the code of my PR which does not change anything related to generated columns and sync_schema()...

This is a serious problem and should be reported

**Comment by fnc12** [comment]
yes it exists. The only solution I see is to know when you should call `sync_schema` and when you should not call it. Also I'll try to find a way how to get `PRAGMA table_info` with generated columns. Thanks @juandent

**Comment by fnc12** [comment]
looks like `PRAGMA table_xinfo` does it. https://sqlite.org/forum/info/de21e54db57102f8

**Comment by fnc12** [comment]
Let's keep it opened until it is fixed in dev

**Comment by juandent** [comment]
ok
let's hope it is soon!!

**Comment by fnc12** [comment]
I guess it works in `dev` so we can close it
